frontend/src/components/grantaccess/GrantAccess.vue

```diff
@@ -220,32 +128,28 @@ const handleSubmit = async () => {
         errorForestClientIdList: [] as string[],
     };
     do {
-        const item = forestClientData.value.pop();
-        const itemForestClientNumber = item?.forest_client_number
-            ? item.forest_client_number
-            : '';
-        const data = toRequestPayload(formData.value, item);
-
+        const forestClientNumber = formData.value.verifiedForestClients.pop();
+        const data = toRequestPayload(formData.value, forestClientNumber);
         await AppActlApiService.userRoleAssignmentApi
             .createUserRoleAssignment(data)
             .then(() => {
-                successForestClientIdList.push(itemForestClientNumber);
+                successForestClientIdList.push(forestClientNumber || '');
```

Comment:
Hi @MCatherine1994 , not sure if this is from previous design issue with subsequent function using the "...IdList", but I am wondering if `forestClientNumber` is "undefined" why does it needs to be pushed into the list with empty string?
Is it because the empty string is being used for output combined message?

Reply:
I was a little bit confused about the logic here as well yesterday. But I think currently the setGrantAccessNotificationMsg method can detect the empty sting in the ForestClientIdList and return the correct message, it has been treated as a special case. I want a refactoring later to make this logic better. But for this ticket, I just follow what we have right now so it won't have too many logic changes.